names/names.py

```diff
@@ -10,13 +10,17 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
-duplicates = []  # Return the list of duplicates in this data structure
+# duplicates = [name for name in names_1 if name in names_2]  # Return the list of duplicates in this data structure
+
+# # Replace the nested for loops below with your improvements
+# for name_1 in names_1: #O(n)
+#     for name_2 in names_2: #O(n)
+#         if name_1 == name_2:
+#             duplicates.append(name_1)
+
+duplicates = set(names_1).intersection(names_2)
```

Comment:
Make sure you're redoing this problem with a different data structure. For MVP it says `You may not use the built in Python list, set, or dictionary for this problem
`.